extensions/panache/mongodb-panache/deployment/src/main/java/io/quarkus/mongodb/panache/deployment/PanacheResourceProcessor.java

```diff
@@ -344,7 +370,7 @@ private void extractMappings(Map<String, String> classPropertyMapping, ClassInfo
         }
 
         // climb up the hierarchy of types
-        if (!target.superClassType().name().equals(DOTNAME_OBJECT)) {
+        if (!target.superClassType().name().equals(io.quarkus.panache.common.deployment.JandexUtil.DOTNAME_OBJECT)) {
```

Comment:
nitpick: We could maybe have an import of `DOTNAME_OBJECT ` and avoid this long line

Reply:
Or we could just import `JandexUtil` since it seems to be used everywhere

Reply:
There is two JandexUtil, one from core deployment and one from Panache.
I think we need to rename the panache one PanacheJandexUtil to be able to import it.
I'll need @FroMage blessing on this rename as he is the original author ;)

Reply:
Okay. Can be done in another PR I suppose. Not a biggie.

Reply:
If I get @stuartwdouglas 's blessing I would move most of our JandexUtil's methods into the deployment one, because it has so many useful methods for everyone, dealing with producing ASM from jandex-using code, that I'm sure other extensions had to reimplement a lot of it.

Reply:
Sounds reasonable.

Reply:
OK thanks. I'll open a mail discussion before doing that PR, to make sure other `JandeUtil` classes don't already exist.